### DISCUSSION TITLE
big And / big Or of a list

### DISCUSSION THREAD

**Discussion by yangyeileen** [QuMuLab/bauhaus#94] [Ideas]
Hey everyone,

Is there an easy way to create a big And/Or of a list of variables? I.e.

`fluents = []`
` term = And(fluents) >> x_a`

Thanks!

**Comment by karishmadaga** [comment]
yes! just create your list of variables and you'll be able to wrap an And or Or around it

creating 'term' with what bauhaus currently supports would have you do the following:
1. create the fluent variables
2. term = ``Or({And(fluents).negate(), x_a})`` which is the NNF form of the implication

**Reply by haz** [reply]
But what must `x_a` and the `fluents` be? NNF Vars? We'd want to do this on the `@proposition` objects (so pre-`compile()`)

**Reply by haz** [reply]
@KarishmaDaga Just went back through the code again -- I don't think we've wrapped any such functionality. Idea here would be to define non-`python-nnf` versions of `And` and `Or` that is just a thin wrapper of `CustomNNF` of the right type. Key thing to keep in mind is that at the time of creating these constraints, there is no `python-nnf` theories/variables created yet.

Any issue you see with the above? I can put a quick PR in for it, if not.

**Reply by karishmadaga** [reply]
sorry my github notifs have been all over the place! yes this makes sense, and nice PR!
